src/btree/bt_handle.c

```diff
@@ -689,21 +689,21 @@ __btree_page_sizes(WT_SESSION_IMPL *session)
 		    "size (%" PRIu32 "B)", btree->allocsize);
 
 	/*
-	 * When a page is forced to split, we want at least 50 entries on its
-	 * parent.
-	 *
 	 * Don't let pages grow larger than a quarter of the cache, with too-
 	 * small caches, we can end up in a situation where nothing can be
 	 * evicted.  Take care getting the cache size: with a shared cache,
 	 * it may not have been set.
 	 */
 	WT_RET(__wt_config_gets(session, cfg, "memory_page_max", &cval));
-	btree->maxmempage =
-	    WT_MAX((uint64_t)cval.val, 50 * (uint64_t)btree->maxleafpage);
+	btree->maxmempage = cval.val;
 	if (!F_ISSET(conn, WT_CONN_CACHE_POOL)) {
-		if ((cache_size = conn->cache_size) > 0)
+		if ((cache_size = conn->cache_size) > 0) {
+			btree->maxmempage =
+			    WT_MIN(btree->maxmempage, cache_size / 10);
+			/* Use a lower bound of a single disk leaf page */
 			btree->maxmempage =
-			    WT_MIN(btree->maxmempage, cache_size / 4);
+			    WT_MAX(btree->maxmempage, btree->maxleafpage);
```

Comment:
@agorrod, I don't disagree with the change, but shouldn't we apply the `maxleafpage` lower bound outside this condition about cache size?